packages/react-components/src/components/table-output-component.tsx

```diff
@@ -101,23 +111,131 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
+        return itemPropsObj;
+    }
+
+    private onKeyDown(event: CellKeyDownEvent) {
+        const gridApi = event.api;
+        const keyEvent = event.event as KeyboardEvent;
+        const rowIndex = event.rowIndex;
+        this.enableIndexSelection = keyEvent.shiftKey;
+
+        if (gridApi && this.timestampCol) {
+            const currentRow = gridApi.getRowNode(String(rowIndex));
+            const currTimestamp = event.data[this.timestampCol];
+
+            let isContiguous = true;
+            if (keyEvent.shiftKey) {
+                if (keyEvent.code === 'ArrowDown') {
+                    if (!currentRow.isSelected()) {
+                        gridApi.deselectAll();
+                        isContiguous = false;
+                    }
+
+                    const nextRow = gridApi.getRowNode(String(rowIndex + 1));
+                    if (nextRow && nextRow.id) {
+                        nextRow.setSelected(true);
+                    }
+
+                    if (isContiguous === false) {
+                        this.selectStartIndex = nextRow.rowIndex;
+                        this.startTimestamp = nextRow.data[this.timestampCol];
+                    }
+
+                    this.selectEndIndex = Math.max(this.selectEndIndex, nextRow.rowIndex);
+                    this.endTimestamp = Math.max(this.endTimestamp, nextRow.data[this.timestampCol]);
+                    this.handleRowSelectionChange();
+                } else if (keyEvent.code === 'ArrowUp') {
+                    if (!currentRow.isSelected()) {
+                        gridApi.deselectAll();
+                        isContiguous = false;
+                    }
+
+                    const nextRow = gridApi.getRowNode(String(rowIndex - 1));
+                    if (nextRow && nextRow.id) {
+                        nextRow.setSelected(true);
+                    }
+
+                    if (isContiguous === false) {
+                        this.selectEndIndex = nextRow.rowIndex;
+                        this.endTimestamp = nextRow.data[this.timestampCol];
+                    }
+
+                    this.selectStartIndex = Math.min(this.selectStartIndex, nextRow.rowIndex);
+                    this.startTimestamp = Math.min(this.startTimestamp, nextRow.data[this.timestampCol]);
+                    this.handleRowSelectionChange();
+                } else if (keyEvent.code === 'Space') {
+                    this.setSelectedRange(rowIndex, currTimestamp);
+                    if (this.startTimestamp === this.endTimestamp) {
+                        gridApi.deselectAll();
+                        if (currentRow && currentRow.id) {
+                            currentRow.setSelected(true);
+                        }
+                    }
+                    this.selectRowsByIndex();
+                    this.handleRowSelectionChange();
+                }
+            } else if (keyEvent.code === 'Space') {
+                gridApi.deselectAll();
```

Comment:
There are a few places like here where there is code that you would expect to execute (related to index selection) even if the timestampCol is not defined. We should be careful to only make timestamp handling conditional on that variable.